test/unit/functions/validation/validFunctionAssignments.spec.ts

```diff
@@ -248,3 +263,21 @@ test("Does not fail on union type signatures (#896)", () => {
         )
         .expectToHaveNoDiagnostics();
 });
+
+// https://github.com/TypeScriptToLua/TypeScriptToLua/issues/1568
+test("No false positives when using generic functions (#1568)", () => {
+    util.testModule`
+        /** @noSelf */
+        declare namespace Test {
+            export function testCallback<T extends (...args: any[]) => void>(
+                callback: T,
+            ): void;
+            export function testCallback2(
+                callback: (...args: any[]) => void,
+            ): void;
+        }
+
+        Test.testCallback(() => {});
+        Test.testCallback2(() => {});
```

Comment:
Could we add a test for:

```
const x = () => {};

Test.testCallback(x);
```

as this also fails on the current version:

https://typescripttolua.github.io/play#code/PQKhAIAEDsHsGUCmAbAZuEwBQATRBjZAQwCdFxoiBbRAZwAcj9yAVOgF3AG8txxEAHvVglOqAK7R87AJaxo4dhwDCRZMgBGTANYAeFvwFLoOWuAAUAOmukA5rQBc4ItACeAbQC6ASnABeAD5wADdYGRwA814+cHw1TR0nFgAaaO8nF1cAbiwAXywsfHlaTgF-C19A7lycthLLJRLVdS18bXMBbyygA